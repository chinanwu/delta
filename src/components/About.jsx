import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import './About.less';
import getThemeClassname from '../functions/getThemeClassname';

export const About = ({ dark }) => {
	useEffect(() => {
		document.title = 'About - Mairead';
	}, []);

	return (
		<div className={getThemeClassname('About', dark)}>
			<div className="About__nav" role="navigation">
				<Link className={getThemeClassname('About__navBtn', dark)} to="/">
					Home
				</Link>
				<Link
					className={getThemeClassname('About__navBtn--disabled', dark)}
					to="/about"
				>
					About
				</Link>
				<Link
					className={getThemeClassname('About__navBtn', dark)}
					to="/settings"
				>
					Settings
				</Link>
			</div>
			<div className="About__header" role="heading" aria-level="1">
				About
			</div>
			<div id="aboutRulesLabel" className="About__label" aria-level="2">
				Rules
			</div>
			<div
				className="About__content"
				role="document"
				tabIndex="0"
				aria-labelledby="aboutRulesLabel"
				aria-level="3"
			>
				In each game, you are given two words, a "from" word and a "to" word.
				Starting from the "from" word, one letter is changed at a time until the
				word becomes the "to" word. However, each time a letter is swapped out,
				the resulting new word must still be a valid four-letter word in the
				English language.
			</div>
			<div id="aboutExampleLabel" className="About__label" aria-level="2">
				Example
			</div>
			<div
				className="About__content"
				role="document"
				tabIndex="0"
				aria-labelledby="aboutExampleLabel"
				aria-level="3"
			>
				From: "heat" -> To: "cold"
				<div role="list" aria-level="4">
					{['heat', 'head', 'held', 'hold', 'cold'].map(word => (
						<div key={word} role="listitem" aria-level="4">
							{word}
						</div>
					))}
				</div>
			</div>
			<div id="aboutThankYouLabel" className="About__label" aria-level="2">
				Big thank you to M.C.S.
			</div>
			<div
				className="About__content"
				role="document"
				tabIndex="0"
				aria-labelledby="aboutThankYouLabel"
				aria-level="3"
			>
				Originator of Mairead, the game
			</div>
		</div>
	);
};

About.propTypes = {
	dark: PropTypes.bool,
};

export const mapStateToProps = ({ theme }) => ({
	dark: theme.dark,
});

export default connect(mapStateToProps)(About);
