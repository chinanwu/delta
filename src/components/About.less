@import '../style/styles';

.About {
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
	height: 100%;
}

.About__nav {
	@nav();
}

.About__navBtn {
	@navBtn();
}

.About__navBtn:hover {
	@navBtnHover();
}

.About__navBtn:active {
	@navBtnActive();
}

.About__header {
	@header();
}

.About__label {
	font-size: 32px;
	margin-bottom: 12px;
}

.About__content {
	text-align: center;
	margin-bottom: 18px;
}

.About__list {
	list-style-type: none;
}
