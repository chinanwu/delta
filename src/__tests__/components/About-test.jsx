import { shallow } from 'enzyme';
import React from 'react';

import { About, mapStateToProps } from '../../components/About.jsx';

jest.unmock('../../components/About.jsx');

describe('About component', () => {
	describe('rendering', () => {
		it('renders About', () => {
			const wrapper = shallow(<About />);
			expect(wrapper.getElement()).toMatchSnapshot();
		});
	});

	describe('mapStateToProps', () => {
		it('returned mapped properties', () => {
			const theme = { dark: true };
			const state = { theme };
			expect(mapStateToProps(state)).toEqual(theme);
		});
	});
});
