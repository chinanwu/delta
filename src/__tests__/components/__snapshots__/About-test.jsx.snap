// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component rendering renders About 1`] = `
<div>
  <Memo(Connect(NavBar))
    activeTab="about"
  />
  <div
    aria-level="1"
    className="About__header"
    role="heading"
  >
    About
  </div>
  <div
    aria-level="2"
    className="About__label"
    id="aboutRulesLabel"
  >
    Rules
  </div>
  <div
    aria-labelledby="aboutRulesLabel"
    aria-level="3"
    className="About__content"
    role="document"
    tabIndex="0"
  >
    In each game, you are given two words, a "from" word and a "to" word. Starting from the "from" word, one letter is changed at a time until the word becomes the "to" word. However, each time a letter is swapped out, the resulting new word must still be a valid four-letter word in the English language.
  </div>
  <div
    aria-level="2"
    className="About__label"
    id="aboutExampleLabel"
  >
    Example
  </div>
  <div
    aria-labelledby="aboutExampleLabel"
    aria-level="3"
    className="About__content"
    role="document"
    tabIndex="0"
  >
    From: "heat" -&gt; To: "cold"
    <div
      aria-level="4"
      role="list"
    >
      <div
        aria-level="4"
        role="listitem"
      >
        heat
      </div>
      <div
        aria-level="4"
        role="listitem"
      >
        head
      </div>
      <div
        aria-level="4"
        role="listitem"
      >
        held
      </div>
      <div
        aria-level="4"
        role="listitem"
      >
        hold
      </div>
      <div
        aria-level="4"
        role="listitem"
      >
        cold
      </div>
    </div>
  </div>
  <div
    aria-level="2"
    className="About__label"
    id="aboutThankYouLabel"
  >
    Big thank you to M.C.S.
  </div>
  <div
    aria-labelledby="aboutThankYouLabel"
    aria-level="3"
    className="About__content"
    role="document"
    tabIndex="0"
  >
    Originator of Mairead, the game
  </div>
</div>
`;
